### ISSUE TITLE
[no-magic-numbers] - default parameter values

### ISSUE THREAD

**Issue by avalanche1** [eslint/eslint#13201]
**Tell us about your environment**

* **ESLint Version:**
^7.0.0-alpha.2

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function foo(param = 3000) {
  // code
}
```

**What did you expect to happen?**
`eslint` will not complain
**What actually happened? Please include the actual, raw output from ESLint.**
`No magic number: 3000`

I think magic numbers in default values for function params either:
- should not be reported
- provide an option to the rule to omit those

The whole point of default values is to specify the value, is it not?

**Comment by anikethsaha** [comment]
I think it does make sense to ignore default arguments of a function as a separate options. 
+1 for this

**Comment by mdjermanovic** [comment]
I think this is correct default behavior, but an option to omit those makes sense to me.

There is already proposal/PR #12611 that includes default values for function params, as clarified in the discussion.

@avalanche1 would you prefer a separate option for function params only?

**Comment by avalanche1** [comment]
#12611 looks fine by me.
actually I wan't aware of default params with destructuring assignment
